lib/dotcom_web/plugs/path_params_redirector.ex

```diff
@@ -0,0 +1,39 @@
+defmodule DotcomWeb.Plugs.PathParamsRedirector do
+  import Plug.Conn, only: [put_status: 2, halt: 1]
+  import Phoenix.Controller, only: [redirect: 2]
+
+  @behaviour Plug
+
+  @impl true
+
+  @spec init(Keyword.t()) :: Keyword.t()
+  def init(opts) do
+    opts
+    |> validate_to_keyword()
+  end
+
+  @impl true
+  @spec call(Plug.Conn.t(), Keyword.t()) :: Plug.Conn.t()
+  def call(conn, to: to) do
+    new_path =
+      to
+      |> add_path_params(conn)
+      |> append_query_string(conn)
+
+    conn
+    |> put_status(:moved_permanently)
+    |> redirect(to: new_path)
+    |> halt()
+  end
+
+  defp add_path_params(to, %Plug.Conn{path_params: %{"path_params" => path_params}}),
```

Comment:
Might want to just make this multi-line:

```
def .... do
  ...
end
```

Reply:
Yeah - I like it when the simple single-line implementations can just get stacked together, but that already wasn't happening here :(

[PTALA](https://github.com/mbta/dotcom/pull/2432/commits/ef6710843655372b184501766f002ad20c38022a)